### ISSUE TITLE
Unexpected behavior with ':version' via `oex -s`

### ISSUE THREAD

**Issue by kajukenbo** [johnsonjh/OpenVi#38]
Given:
`$ printf ':version\n:set\n:q!\n' | ex -s`
```
 HP Version $ B.11.31  Oct  1 2008 02:03:25 $
noautoprint optimize noprompt term=dumb
```

And:
`$ printf ':version\n:set\n:q!\n' | ex -s`
```
Version 4.0 (gritter) 3/25/05
noautoprint noprompt term=dumb
```

Compare to:
`$ printf ':version\n:set\n:q!\n' | oex -s`
`noautoprint     noprompt        term="xterm"    nowarn`

FWIW, this is probably some old `nvi` bug:
`$ printf ':version\n:set\n:q!\n' | nex -s`
`noautoprint     noprompt        term="xterm"    nowarn`

**Comment by johnsonjh** [comment]
Vim's `ex` behaves like nvi and OpenVi.  AIX's behaves like the traditional ones, as expected.  The nvi source specifically sets a message of M_INFO level, which is suppressed when in ex silent mode.

So, while it appears this is intentional behavior, it's not matching traditional vi, so I don't mind changing it.   I'm not sure using an error-level output makes sense here ... it would do the right thing but it isn't really an error (nor would I ever want or expect it output to stderr.)

I think using *ex_printf* here should be side-effect free.

I made this change in 7aa58ab2b4443637cbb1dccae7f46d9c47c2fa9a.

Please let me know if this isn't a good solution and we can reopen.

**Comment by kajukenbo** [comment]
@johnsonjh 

Hi Jeffrey.

I think that is a reasonable solution.
I would presume the "gritter" code does something similar.

Thank you for looking into it.

**Comment by johnsonjh** [comment]
@kajukenbo 

I actually don't like this because, in visual mode, `:abbrev|version` gives:
```
No abbreviations to displayVersion 7.5.29-dev (OpenVi) 04/25/2024.
```

... as there isn't a built-in level ignoring silent mode, so I added one, and use it in fe245de8c702a70cf73528a6a625f4290e82d607.  This will fix that.

This new level allows the version message to stack, where something like ex_puts would not, which I like.  This way, your test case works like you want, and cases like `vi -C version` looks nice too:

```
Version 7.5.29-dev (OpenVi) 04/25/2024; /tmp/vi.1saXWO: new file: line 1
```

Let me know if this works better!  I also noticed I forgot to update the usage display for ex, when I added -C, so I'll fix that too.

Thanks for reporting.
